### ISSUE TITLE
Make the telephone property more structured (not just Text)

### ISSUE THREAD

**Issue by pierrefar** [schemaorg/schemaorg#188]
http://schema.org/telephone

http://schema.org has machine-readable DataType values for dates, prices, etc. The schema.org/telephone property has an expected value of Text, which is freeform and ambiguous.

It would probably better to have the expected value be stricter, following one of the standard phone formats like E.123 or similar:

http://en.wikipedia.org/wiki/E.123

**Comment by cp5w** [comment]
+1 on this suggestion and surprised its not already part of schema.org

@danbri any movement on this? The thread seems to have gone quiet.

Would `Telephone` be a sublcass of `Thing` > `Intangible` > `StructuredValue` or a `DataType` in its own right (like `URL` is)?

**Comment by dbs** [comment]
@cp5w I don't think we need a schema:Telephone type, we would just want to add "URL" to rangeIncludes for the schema:telephone property and adjust existing examples / add some examples.

+1 for adding URL to the range of schema:telephone so that tel: URIs are explicitly supported and encouraged. (I'm already using them, and occasionally even naughty ones that include a space which then raises an exception in Python rdflib... such is the dirty, dirty nature of the web.)

**Comment by mfhepp** [comment]
+1 for adding URL to the range of schema:telephone so that tel: URIs are explicitly supported

**Comment by RichardWallis** [comment]
+1

If applied, recommend adjusting at least one of the current examples to show this.
eg.

From:
```html
  <span property="telephone">(425) 123-4567</span>
  <a href="mailto:jane-doe@xyz.edu" property="email">
```
To:
```html
  <a href="tel:+1-425-123-4567" property="telephone">(425) 123-4567</a>
  <a href="mailto:jane-doe@xyz.edu" property="email">
```

**Comment by scor** [comment]
+1 on adjusting the example and allowing URL as range for telephone. Created pull request https://github.com/schemaorg/schemaorg/pull/757

**Comment by unor** [comment]
The suggestion to add `URL` to the expected range (for using `tel` URIs) got a few +1s and no objections here. The PR #757 contains the needed changes.

Is there anything that needs to be done before this could be committed?

**Comment by danbri** [comment]
@unor this seems sensible - but I think my comment in the associated issue remains underexplored:

> Do we know what browsers (desktop, mobile) do with phone numbers? let's check as rdfa/microdata markup on links is relevant there...

(also note to self, don't lazily merge the PR as it'll change an old branch)

**Comment by unor** [comment]
@danbri Can you clarify what you mean? 

Authors that currently use `tel` URIs and make use of Schema.org’s `telephone` property typically would have markup like:

```html
<a href="tel:+0-123"><span property="telephone">(0) 123</span></a>
```

When these authors switch to a URL value, they would have something like:

```html
<a property="telephone" href="tel:+0-123">(0) 123</a>
```

So nothing about what user agents do should change here (`property`/`itemprop` don’t affect how the hyperlink works, bugs aside).

Or do you mean authors that start using hyperlinks because URL becomes an expected value, but who might not be aware how `tel` hyperlinks work for their users (i.e., opening a suitable phone client, which is typically installed on mobile devices, but not on desktop devices)? I would prefer to let this be the author’s responsibility.  
Maybe we should show an example that uses text, an example that uses `a`, and an example that uses `link`? So as not to imply that authors *should* use `a`.

**Comment by thadguidry** [comment]
As a reference, here is what Skype URI looks like

`<a href="skype:echo123?call">Call the Skype Echo / Sound Test Service</a>`

https://msdn.microsoft.com/library/office/dn745878
https://msdn.microsoft.com/en-us/library/office/dn745882

**Comment by thadguidry** [comment]
@danbri Firefox has several tests for the TEL URI has shown here: https://github.com/mozilla/gecko-dev/search?utf8=%E2%9C%93&q=%22tel%3A%22

Telephone URL protocal defined here: https://www.ietf.org/rfc/rfc2806.txt

Firefox's JS handler is coded here: https://github.com/mozilla/gecko-dev/blob/86897859913403b68829dbf9a154f5a87c4b0638/b2g/components/TelProtocolHandler.js

**Comment by VladimirAlexiev** [comment]
+1 for **not** introducing a `Telephone` datatype, a URL with `tel:` scheme is all we need

**Comment by westurner** [comment]
Another :+1: for changing the range of .telephone to {Text, URL} to support `:tel:` URIs

**Comment by westurner** [comment]
Is this where `:rangeIncludes` should be changed to `:URL, :Text`?

https://github.com/schemaorg/schemaorg/blob/95d0e0cb29c9f4622a045fe8a9c3a692e1d591ce/data/schema.ttl#L8914-L8921

Which examples should be upgraded to the new URL format? 

- [ ] https ://github.com/schemaorg/schemaorg/blob/main/data/sdo-dentist-examples.txt
- [ ] ?

**Comment by erik-ros-devilsberg** [comment]
I guess this is not a priority for anyone, but if we are ever changing the telephone field in schema.org markups, we should also create space for the type of telephone number (home/ mobile/ work, etc) (imo).

**Comment by mfhepp** [comment]
> I guess this is not a priority for anyone, but if we are ever changing the telephone field in schema.org markups, we should also create space for the type of telephone number (home/ mobile/ work, etc) (imo).

I think [schema:ContactPoint](https://schema.org/ContactPoint) is sufficient for this, because you can e.g. use the property [schema:contactType](https://schema.org/contactType) for indicating "home" vs. "cell" etc.

From a more academic standpoint, the **role** that a certain telephone number has may be dependent on the temporal context and other contextual parameters. My home phone number could be my friends' secondary phone number, or be the office phone number for a start-up operated by one of my family members etc. :-).

A very worthwhile, yet not casual, reading for anybody working on shared data structures is the [OntoClean](https://en.wikipedia.org/wiki/OntoClean) methodology by Guarino and Welty (more comprehensive, yet paywalled is this [bookchapter on OntoClean](https://link.springer.com/chapter/10.1007/978-3-540-24750-0_8)). While much broader in its practical impact, it tells us that it is important to distinguish between **rigid** properties of something and its **non-essential** properties, e.g. **roles** and other aspects that e.g. may change or cease to exist without the need for the main entity to cease to exist.

Put a bit simpler: The relationship between an agent (e.g. a person) and a telephone number is an N-ary one. Still relevant for schema.org in JSON-LD times is e.g. [Hayes/Welty: *Defining N-ary Relations on the Semantic Web*](https://www.w3.org/TR/swbp-n-aryRelations/).

Much problems in very pragmatic IT scenarios would be non-issues if data structures would not mix e.g. a person's stable characteristics from the ephemeral ones tied to certain roles or contexts.

This is not at all a rant :-) and a bit off-topic, I know ;-).

**Comment by erik-ros-devilsberg** [comment]
I was a bit quick with my comment, Contact point is indeed a solution, albeit a bit over engineered for my needs. thank you for sharing those links, I will defo read up! 
I guess I have nothing more to say about this without getting philosophical or political. keep up the good work!

**Comment by gmackenz** [comment]
As part of a general cleanup of long oustanding issues, this one appears to have no consensus for changes or no changes/use existing schema—closing this issue for nowl
